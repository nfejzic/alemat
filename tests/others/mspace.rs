use alemat::{
    attributes::Attribute,
    elements::{Frac, Num, Space, SpaceAttr},
    MathMl,
};

#[test]
fn mspace() {
    let out = MathMl::with_content(alemat::children![
        Num::from(1),
        Space::default().with_attr([
            SpaceAttr::Width("1em".into()),
            SpaceAttr::Global(Attribute::Style("border-top: 1px solid blue".into()))
        ]),
        Frac::builder()
            .num(alemat::row![
                Num::from(2),
                Space::default().with_attr([
                    SpaceAttr::Depth("1em".into()),
                    SpaceAttr::Global(Attribute::Style("border-left: 1px solid blue".into()))
                ]),
            ])
            .denom(alemat::row![
                Num::from(3),
                Space::default().with_attr([
                    SpaceAttr::Height("2em".into()),
                    SpaceAttr::Global(Attribute::Style("border-left: 1px solid blue".into()))
                ]),
            ])
            .build()
    ])
    .render();

    crate::snap_test!(out, name: "others_mspace");
}
