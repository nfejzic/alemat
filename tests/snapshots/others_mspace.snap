---
source: tests/others/mspace.rs
expression: out
---
<math><mn>1</mn><mspace width="1em" style="border-top: 1px solid blue"/><mfrac><mrow><mn>2</mn><mspace depth="1em" style="border-left: 1px solid blue"/></mrow><mrow><mn>3</mn><mspace height="2em" style="border-left: 1px solid blue"/></mrow></mfrac></math>
